src/app/workbench/useLoadViewPlugin.ts

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import {AView, EXTENSION_POINT_TDP_VIEW, IViewPlugin, IViewPluginDesc, LoginMenu, PluginRegistry, useAsync, Range, IView, ObjectRefUtils, ResolveNow, IDType, LocalStorageProvenanceGraphManager, ARankingView, IDTypeManager, FindViewUtils, IDiscoveredView} from 'tdp_core';
+import {addTransitionOptions, useAppDispatch, useAppSelector} from '../..';
+import {getAllFilters} from '../../store/storeUtils';
+
+
+export function useLoadViewPlugin(viewId: string, workbenchIndex: number): [(element: HTMLElement | null) => void, IView | null] {
+    const view = PluginRegistry.getInstance().getPlugin(EXTENSION_POINT_TDP_VIEW, viewId) as IViewPluginDesc;
+
+
+    const dispatch = useAppDispatch();
+    const ordino = useAppSelector((state) => state.ordino);
+    const [instance, setInstance] = React.useState<IView | null>(null);
+    const loadView = React.useMemo(() => () => {
+        return view.load();
+    }, []);
+
+
+    const {status, value: viewPlugin} = useAsync(loadView, []);
+
+    const setRef = React.useCallback(async (ref: HTMLElement | null) => {
+        // Create a new one if there is a ref
+        if (ref && status === 'success') {
+
+            const idType = workbenchIndex === 0 ? 'Start' : ordino.workbenches[workbenchIndex - 1].entityId;
+
+            const selection = {idtype: new IDType(idType, viewId, '', true), range: workbenchIndex === 0 ? Range.none() : Range.list(ordino.workbenches[workbenchIndex - 1].selections)};
```

Comment:
Refactoring suggestion:

```suggestion
            const previousWorkbench = ordino.workbenches[workbenchIndex - 1];
            
            const idType = (previousWorkbench) ? previousWorkbench.entityId : Start;
            const range = (previousWorkbench) ? Range.list(previousWorkbench.selections) : Range.none();
            
            const selection = {idtype: new IDType(idType, viewId, '', true), range};
```